modules/calcite/src/test/java/org/apache/ignite/internal/processors/query/calcite/integration/KillCommandDdlIntegrationTest.java

```diff
@@ -0,0 +1,286 @@
+
+            assertThrowsWithCause(() -> fut.get(TIMEOUT), IgniteException.class);
+        } finally {
+            computeLatch.countDown();
+        }
+    }
+
+    /** */
+    @Test
+    public void testCancelTx() {
+        IgniteCache<Object, Object> cache = client.cache(DEFAULT_CACHE_NAME);
+
+        int testKey = PAGES_CNT * (PAGE_SZ + 1);
+
+        try (Transaction tx = client.transactions().txStart()) {
+            cache.put(testKey, 1);
+
+            executeSql(client, "KILL TRANSACTION '" + tx.xid() + "'");
+
+            assertThrowsWithCause(tx::commit, IgniteException.class);
+        }
+
+        assertNull(cache.get(testKey));
+    }
+
+    /** @throws Exception If failed. */
+    @Test
+    public void testCancelService() throws Exception {
+        String serviceName = "MY_SERVICE";
+
+        ServiceConfiguration scfg = new ServiceConfiguration();
+        scfg.setName(serviceName);
+        scfg.setMaxPerNodeCount(1);
+        scfg.setNodeFilter(grid(0).cluster().predicate());
+        scfg.setService(new TestServiceImpl());
+
+        client.services().deploy(scfg);
+
+        TestService svc = client.services().serviceProxy(serviceName, TestService.class, true);
+        assertNotNull(svc);
+
+        executeSql(client, "KILL SERVICE '" + serviceName + "'");
+        boolean res = waitForCondition(() -> grid(0).services().serviceDescriptors().isEmpty(), TIMEOUT);
+        assertTrue(res);
+    }
+
+    /** @throws Exception If failed. */
+    @Test
+    public void testCancelContinuousQuery() throws Exception {
+        IgniteCache<Object, Object> cache = client.cache(DEFAULT_CACHE_NAME);
+
+        ContinuousQuery<Integer, Integer> cq = new ContinuousQuery<>();
+
+        AtomicInteger cntr = new AtomicInteger();
+
+        cq.setInitialQuery(new ScanQuery<>());
+        cq.setTimeInterval(1_000L);
+        cq.setPageSize(PAGE_SZ);
+        cq.setLocalListener(events -> {
+            for (CacheEntryEvent<? extends Integer, ? extends Integer> e : events) {
+                assertNotNull(e);
+
+                cntr.incrementAndGet();
+            }
+        });
+
+        cache.query(cq);
+
+        for (int i = 0; i < PAGE_SZ * PAGE_SZ; i++)
+            cache.put(i, i);
+
+        boolean res = waitForCondition(() -> cntr.get() == PAGE_SZ * PAGE_SZ, TIMEOUT);
+        assertTrue(res);
+
+        Map<UUID, Object> routines = getFieldValue(grid(0).context().continuous(), "rmtInfos");
+        assertEquals(1, routines.size());
+        Map.Entry<UUID, Object> entry = routines.entrySet().iterator().next();
+
+        UUID nodeId = getFieldValue(entry.getValue(), "nodeId");
+        UUID routineId = entry.getKey();
+
+        executeSql(client, "KILL CONTINUOUS '" + nodeId + "' '" + routineId + "'");
+
+        long cnt = cntr.get();
+
+        for (int i = 0; i < PAGE_SZ * PAGE_SZ; i++)
+            cache.put(i, i);
+
+        res = waitForCondition(() -> cntr.get() > cnt, TIMEOUT);
```

Comment:
Too large timeout to wait for something not happening.
Let's just wait for CQ system view become empty and check the counter after that:
```
        assertTrue(waitForCondition(() -> F.isEmpty(qrys), TIMEOUT));
        assertEquals(cnt, cntr.get());
```